src/basic/random-util.c

```diff
@@ -248,3 +249,34 @@ uint64_t random_u64_range(uint64_t m) {
 
         return x % m;
 }
+
+/* Returns 1 if the seed file is okay, 0 if not, and < 0 on error determining. */
+int verify_random_seed_file_permissions(int seed_fd) {
+        _cleanup_free_ char *full_path = NULL, *prefix = NULL;
+        struct stat st;
+        int r;
+
+        if (fstat(seed_fd, &st) < 0)
+                return log_error_errno(errno, "Unable to fstat random seed file descriptor: %m");
+        if (!S_ISREG(st.st_mode))
+                return log_error_errno(EBADE, "Random seed file is not a regular file");
+        if (!(st.st_mode & 0007))
+                return 1;
+
+        r = fd_get_path(seed_fd, &full_path);
+        if (r < 0)
+                return log_error_errno(r, "Unable to determine full path of random seed file: %m");
```

Comment:
hmm, to give this a less racy/symlink-vulnerable feel (not that it would matter much, but still), maybe code this like this:

```c
r = path_extract_directory(full_path, &parent);
if (r < 0) 
        …

dfd = open(parent, O_DIRECTORY|O_CLOEXEC);
if (dfd < 0)
        …

if (fstat(&dfd, &st) < 0)
        … 

while (;;) {
        _cleanup_close int parent_dfd = -1;
        struct stat parent_st;

        if ((st.st_mode & 00007) != 0) 
                …   
        
        parent_dfd = openat(dfd, "..", O_DIRECTORY|O_CLOEXEC);
        if (parent_dfd < 0)
                …

        if (fstat(parent_dfd, &parent_st) < 0) 
                …

        if (stat_inode_same(&st, &parent_st)) /* reached root dir */
                break;

        close_and_replace(dfd, parent_dfd);
        st = parent_st;
}